### PR TITLE
feat: add latex_figure_context_extractor_mapper operator

### PR DESCRIPTION
**Pull request by liyuyi-2001**
## Summary

Add a new mapper operator `latex_figure_context_extractor_mapper` that extracts figures and their citing context from LaTeX source.

## What it does

- Parses `figure`, `figure*`, `wrapfigure` environments
- Handles `subfigure` environments and `\subfigure`/`\subfloat` commands
- Extracts caption, label, and `\includegraphics` paths for each figure
- Finds prose paragraphs that cite each figure via `\ref`/`\cref`/`\Cref`/`\autoref`
- Fans out one paper row into N figure rows (one per figure/subfigure)
- Drops samples that contain no figures with images

## Output fields

| Field | Type | Description |
|-------|------|-------------|
| `images` | `list[str]` | Image paths from `\includegraphics` |
| `caption` | `str` | Figure caption text |
| `label` | `str` | LaTeX label string |
| `citing_paragraphs` | `list[str]` | Paragraphs that cite this figure |
| `parent_caption` | `str` | Parent figure caption (subfigures only) |
| `parent_label` | `str` | Parent figure label (subfigures only) |

## Changes

- `data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py` — operator implementation
- `data_juicer/ops/mapper/__init__.py` — registration
- `data_juicer/config/config_all.yaml` — config entry
- `tests/ops/mapper/test_latex_figure_context_extractor_mapper.py` — 21 unit tests
- `docs/operators/mapper/latex_figure_context_extractor_mapper.md` — documentation (EN/CN)

## Testing

- 21 unit tests covering: single figures, subfigure environments, subfigure/subfloat commands, nested braces, caption variants, citation matching, label boundaries, custom keys, batch fan-out, and edge cases.

### CODE REVIEW

File: data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+      paragraphs that cite this figure.
+    - ``<parent_caption_key>`` (default ``parent_caption``): parent
+      figure caption (subfigures only; empty for standalone figures).
+    - ``<parent_label_key>`` (default ``parent_label``): parent
+      figure label (subfigures only; empty for standalone figures).
+
+    Note: this operator expects the full LaTeX source as a single
+    string.  It does **not** resolve ``\\input`` or ``\\include``
+    directives.  If your documents span multiple ``.tex`` files,
+    concatenate them into a single text field before applying this
```
Comment (yxdyc) [inline data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py:67]: Can we have a dedicated op for this step? "If your documents span multiple ``.tex`` files,
    concatenate them into a single text field before applying this"

File: data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+                subfig_env_matches + subfig_cmd_matches
+                + subfig_nocap_matches,
+                key=lambda m: m.start(),
+                reverse=True,
+            )
+            for m in all_matches:
+                text_outside = (
+                    text_outside[:m.start()]
+                    + text_outside[m.end():]
+                )
```
[the remark this answers is not in the document]
Comment (liyuyi-2001) [reply data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py]: This is not a bug. The matches are sorted in reverse order (reverse=True) before the loop, so removals proceed right-to-left

File: data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+                            sf.label, paragraphs
+                        )
+                        # Merge parent + subfigure, deduplicated,
+                        # preserving order
+                        seen = set()
+                        merged = []
+                        for p in parent_citing + sf_citing:
+                            if p not in seen:
+                                seen.add(p)
+                                merged.append(p)
```
[the remark this answers is not in the document]
Comment (liyuyi-2001) [reply data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py]: done

File: data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py (hunk 1 above)
Comment (liyuyi-2001) [reply data_juicer/ops/mapper/latex_figure_context_extractor_mapper.py:67]: I added a mapper here, which will load the archive files and concat them. Please consider reviewing it. https://github.com/datajuicer/data-juicer/pull/932

**Comment by liyuyi-2001** [comment]
> plz resolve the gemini suggestion & my minor comment
> 
> others LGTM.

Gemini's valid suggestion has been resolved. I've also added a mapper in https://github.com/datajuicer/data-juicer/pull/932, which might address your comment.
